app/main/posts/views/post-view-list.directive.js

```diff
@@ -97,18 +97,37 @@ function PostListController(
         PostEndpoint.query(postQuery).$promise.then(function (postsResponse) {
             // Add posts to full set of posts
             // @todo figure out if we can store these more efficiently
-            Array.prototype.push.apply($scope.posts, postsResponse.results);
 
+            // TODO/question: feels like this logic should belong somewhere else in the application? But where?
```

Comment:
Yep so, the filtering is complex, it should live here
https://github.com/ushahidi/platform-client/tree/develop/app/main/posts/views/filters